driverapi/driverapi.go

```diff
@@ -42,6 +42,14 @@ type Driver interface {
 	// Leave method is invoked when a Sandbox detaches from an endpoint.
 	Leave(nid, eid string) error
 
+	// ProgramExternalConnectivity invokes the driver method which does the necessary
+	// programming to allow the external connectivity dictated by the passed options
+	ProgramExternalConnectivity(nid, eid string, options map[string]interface{}) error
```

Comment:
I know we had the discussion regarding this pre-1.10. But, I dont recollect the reason why we need this new function ?
Why cant we reuse the Join's Container Options instead of adding a new function and forcing the drivers to implement it ?